### PR TITLE
[date widget] fix current date can't be picked

### PR DESCRIPTION
**Pull request by 3nids**
fix #16579

Current date couldn't be picked up when allowing NULL date in the calendar widget because it was actually the already selected date in the subsequent QDateTimeEdit.

So now, when clearing the `QgsDateTimeEdit` (setting to NULL), the date of `QDateTimeEdit` is set to `mimimumDate()` (but only when in `calendarPopup()`is true).

To display the calendar at the current date page, it detects `QCalendarWidget::selectionChanged` signal, and if the date is the `mimimumDate()` the page is changed to current date.

There is two (very) small drawbacks:
* if one needs to use the `mimimunDate()` value and widget accepts NIULL values, the calendar will show up at today's page and not at the value's page (minimum date)
* turning off then on again the calendar popup will lead to not showing the current page properly (as the calendar widget object will be deleted and thus disconnected).

I think these are fairly acceptable...

I tried a few other ways, and that's the only one I got working properly.

One of the other approach was to subclass QCalendarWidget and implement showEvent but this is useless as the setDate happens afterwards.
I also tried to setCurrentPage of the calendar when clearing the date time edit, but it's the same issue.

### CODE REVIEW

File: src/gui/editorwidgets/qgsdatetimeedit.cpp [hunk 1]
```diff
@@ -73,6 +89,7 @@ void QgsDateTimeEdit::setEmpty()
   mNullLabel->setVisible( false );
   lineEdit()->setVisible( false );
   mClearButton->setVisible( mAllowNull );
+  mIsEmpty = true;
```
Comment (3nids) [inline src/gui/editorwidgets/qgsdatetimeedit.cpp]: @nyalldawson following my comment in 6d868598f5e477c4dfe25a0acd80e6370e291f2b 
is this correct?

Comment (nyalldawson) [reply src/gui/editorwidgets/qgsdatetimeedit.cpp]: Yes - thanks!

**Comment by nyalldawson** [comment]
Just wondering - did you try listening for QCalender::clicked() and activated()? I'm thinking an alternative approach might be to connect to those signals on the calendarWidget(), and put the current date handling in there instead... If it works, it might be a bit less fragile.

**Comment by 3nids** [comment]
I gave it a try without success (connecting QCalendar::activated/clickec to QDateTimeEdit::setDate and removing the set min date).

Removing the minimum date trick leads to another issue: if you clear the widget after having a date selected (let's say yesterday), when you open the calendar afterwards, yesterday is still selected as current date, which isn't great).
Also, the second issue (turning on/off calendar popup) would remain.

Another idea would be to remove the line edit and play with special value text on the min date. We would have to catch clicks on up/down arrows to fix the date to today prior to increase/decrease it.
But this would rather solve the cosmetic issue (#5959) than this one...

**Comment by nyalldawson** [comment]
Ok... happy for this current approach then, if you think it's best.

**Comment by 3nids** [comment]
OK, this now uses special value text at minimum dates to display NULL.
Thus I could remove the additional line edit (and solve the cosmetic issue at the same time).

I think it's in a better shape like this!

@nyalldawson do you want to review?

**Comment by nyalldawson** [comment]
Much cleaner - looks great!

**Pull request merged**